### PR TITLE
Feature/sipps 415 tech debt

### PR DESCRIPTION
**Pull request by slaphead**
Unit tests added and coverage increased

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        });
+    });
+
+    it('should handle errors from the authCode end point', (done) => {
+      const doneFn = done;
+      const client = sipClient.newClient(clientConfig);
+      const errorMessage = 'There was an error';
+
+      nock.cleanAll();
+      mockAuthCodeThrowErrror(authCode, errorMessage);
```
Comment (SyntaxZA) [inline test/index.js]: ```mockAuthCodeThrowError``` requires 1 argument but 2 given. Perhaps pass in errorMessage to your function. Not a breaking change so not a biggy

Comment (SyntaxZA) [reply test/index.js]: Also spelling error :) ```mockAuthCodeThrowErrror```

File: test/index.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    it('should allow a config with optional keys', () => {
+      const client = sipClient.newClient(clientConfig);
+      expect(client.exchangeCode).to.be.a('function');
+    });
+
+    it('should exchange an encrypted code', (done) => {
+      const doneFn = done;
+      const client = sipClient.newClient(clientConfig);
+      client.exchangeCode(authCode).then((data) => {
+        // console.log(data);
```
Comment (SyntaxZA) [inline test/index.js]: Console log cleanup

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          command: npm install
+      - run:
+          name: Lint Test
+          command: npm run lint
+      - save_cache:
+          key: npm-cache-{{ checksum "package.json" }}
+          paths:
+            - ./node_modules
+    
+  publish-package:
```
Comment (SyntaxZA) [inline .circleci/config.yml]: Shouldn't the CircleCi config contain the ```npm publish``` script to publish the new patch version

File: package.json [hunk 4]
```diff
@@ -11,10 +11,10 @@
   "nyc": {
     "check-coverage": true,
     "per-file": true,
-    "lines": 60,
-    "statements": 60,
-    "functions": 50,
-    "branches": 30,
+    "lines": 89,
```
Comment (SyntaxZA) [inline package.json]: Any reason for this exact number or was it just to pass the tests?

File: .circleci/config.yml (hunk 3 above)
Comment (slaphead) [reply .circleci/config.yml]: We are going to leave it for now. We are still figuring out this part with permissions etc. I am not 100% comfortable yet with auto publish

File: test/index.js (hunk 1 above)
Comment (SyntaxZA) [reply test/index.js]: ```mockAuthCodeThrowError```, there is a triple 'r'

File: .circleci/config.yml (hunk 3 above)
Comment (SyntaxZA) [reply .circleci/config.yml]: Makes sense, was just checking 👍

**Pull request merged**